subprojects/snapshots/src/main/java/org/gradle/internal/vfs/VirtualFileSystem.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.vfs;
+
+import org.gradle.internal.snapshot.FileSystemLocationSnapshot;
+import org.gradle.internal.snapshot.FileSystemSnapshotVisitor;
+import org.gradle.internal.snapshot.SnapshottingFilter;
+
+public interface VirtualFileSystem {
+
+    void read(String location, FileSystemSnapshotVisitor visitor);
+
+    void read(String location, SnapshottingFilter filter, FileSystemSnapshotVisitor visitor);
+
+    void update(Iterable<String> locations, Runnable action);
```

Comment:
This `Runnable` will receive some kind of invalidator later, right?

Reply:
Yes, it will. For now, we always invalidate all the locations.